### PR TITLE
HSEARCH-3021 Document the integration to DI frameworks

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3021

### CODE REVIEW

File: documentation/src/main/asciidoc/mapping.asciidoc [hunk 1]
```diff
@@ -2226,6 +2226,34 @@ CustomBridge cb = classLoaderService.classForName( "com.package.CustomBridge" );
...
+and if not found it will default to reflection-based instantiation (using the public, no-argument constructor).
+
+[WARNING]
+====
+Be careful to define the scope of your bridge as appropriate.
+For instance if the bridge implements `AppliedOnTypeAwareBridge`, a singleton bridge will not work,
+since the bridge may be used in multiple places on different properties with different types.
+In that case, a "prototype" (Spring terminology) or "dependent" (CDI terminology) scope is what you want,
+and it's generally the safest choice.
+====
```
Comment (Sanne) [inline documentation/src/main/asciidoc/mapping.asciidoc]: good point! I'm not very familiar with CDI  myself (only used its precursors). Do you think we can actually verify this to be sane at bootstrap?

**Comment by Sanne** [comment]
great,  merged! 

The single comment I had should be treated as a different JIRA - if at all.

File: documentation/src/main/asciidoc/mapping.asciidoc (hunk 1 above)
Comment (yrodiere) [reply documentation/src/main/asciidoc/mapping.asciidoc]: Not cleanly with the current SPIs at my disposal: I don't have any way to know what the scope of a component is. 

For CDI I would have to bypass Hibernate ORM's SPIs to get access to the bean manager, and even then I could not handle Spring.

We may be able to add something to Hibernate ORM's SPIs, but... TBH it's be hard enough to get where we are now :)

Other than that, I guess I could keep a map "bridge type" => "set of created instances so far" and check that a newly created instance is not part of the set of created instances so far, but I think there would be a lot of false negatives, especially with that habit of DI frameworks' to add proxies everywhere.

Comment (Sanne) [reply documentation/src/main/asciidoc/mapping.asciidoc]: ok let's not explore this further, at least until there's demand.

**Pull request merged**